### PR TITLE
Swift 3 Conversion (changes for Vimeo-iOS)

### PR DESCRIPTION
**Pull request by ghking**
IGNORE ANY `Pods` FILES!!!

- DescriptorKVObserver.swift was somehow not included in the example project, so it didn't get converted. This PR adds it and converts it to Swift 3.
- Changes some `public` things to open where they're subclassed in the Vimeo-iOS

### CODE REVIEW

**Comment by cameo-github** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          3 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoUpload/blob/8d3e61bc790a1ad2607bdedc71c5879325e3ca04/VimeoUpload/UIKit/DescriptorKVObserver.swift#L31">VimeoUpload/UIKit/DescriptorKVObserver.swift#L31</a></strong>: extraneous duplicate parameter name; ‘descriptorType’ already has an argument label  <br /><code>
    func descriptorStateDidChange(descriptorType descriptorType: VideoDescriptorType, descriptorState: DescriptorState, isCancelled: Bool, descriptorError: NSError?)
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoUpload/blob/8d3e61bc790a1ad2607bdedc71c5879325e3ca04/VimeoUpload/UIKit/DescriptorKVObserver.swift#L32">VimeoUpload/UIKit/DescriptorKVObserver.swift#L32</a></strong>: extraneous duplicate parameter name; ‘descriptorType’ already has an argument label  <br /><code>
    func descriptorProgressDidChange(descriptorType descriptorType: VideoDescriptorType, progress: Double)
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoUpload/blob/8d3e61bc790a1ad2607bdedc71c5879325e3ca04/VimeoUpload/Upload/Extensions/AVAsset+Extensions.swift#L32">VimeoUpload/Upload/Extensions/AVAsset+Extensions.swift#L32</a></strong>: extraneous duplicate parameter name; ‘completion’ already has an argument label  <br /><code>
    func approximateFileSize(completion completion: @escaping FloatBlock)
</code></td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          2 Messages
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 2 tests, with 0 failures (0 unexpected) in 0.308 (0.310) seconds</td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 2 tests, with 0 failures (0 unexpected) in 0.359 (0.362) seconds</td>
    </tr>
  </tbody>
</table>

See build details on [CircleCI](https://circleci.com/gh/vimeo/VimeoUpload/163)

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: Examples/VimeoUpload-iOS-OldUpload/VimeoUpload-iOS-OldUpload/Cells/UploadCell.swift [hunk 1]
```diff
@@ -196,7 +196,7 @@ class UploadCell: UITableViewCell
             {
             case(type(of: self).ProgressKeyPath, .some(&self.progressKVOContext)):
                 
-                let progress = (change?[NSKeyValueChangeKey.newKey] as AnyObject).doubleValue ?? 0;
+                let progress = (change?[.newKey] as AnyObject).doubleValue ?? 0;
```
Comment (supperspidey) [inline Examples/VimeoUpload-iOS-OldUpload/VimeoUpload-iOS-OldUpload/Cells/UploadCell.swift]: Semicolon in Swift? Nein.

File: Examples/VimeoUpload-iOS-OldUpload/VimeoUpload-iOS-OldUpload/Cells/UploadCell.swift [hunk 2]
```diff
@@ -208,7 +208,7 @@ class UploadCell: UITableViewCell
                 
             case(type(of: self).StateKeyPath, .some(&self.stateKVOContext)):
                 
-                let stateRaw = (change?[NSKeyValueChangeKey.newKey] as? String) ?? DescriptorState.ready.rawValue;
+                let stateRaw = (change?[.newKey] as? String) ?? DescriptorState.ready.rawValue;
```
Comment (supperspidey) [inline Examples/VimeoUpload-iOS-OldUpload/VimeoUpload-iOS-OldUpload/Cells/UploadCell.swift]: Same here.

File: Examples/VimeoUpload-iOS/VimeoUpload-iOS/Cells/VideoCell.swift [hunk 3]
```diff
@@ -229,7 +229,7 @@ class VideoCell: UITableViewCell
             {
             case(type(of: self).ProgressKeyPath, .some(&self.progressKVOContext)):
                 
-                let progress = (change?[NSKeyValueChangeKey.newKey] as AnyObject).doubleValue ?? 0;
+                let progress = (change?[.newKey] as AnyObject).doubleValue ?? 0;
```
Comment (supperspidey) [inline Examples/VimeoUpload-iOS/VimeoUpload-iOS/Cells/VideoCell.swift]: Here too.

File: Examples/VimeoUpload-iOS/VimeoUpload-iOS/Cells/VideoCell.swift [hunk 4]
```diff
@@ -241,7 +241,7 @@ class VideoCell: UITableViewCell
 
             case(type(of: self).StateKeyPath, .some(&self.stateKVOContext)):
                 
-                let stateRaw = (change?[NSKeyValueChangeKey.newKey] as? String) ?? DescriptorState.ready.rawValue;
+                let stateRaw = (change?[.newKey] as? String) ?? DescriptorState.ready.rawValue;
```
Comment (supperspidey) [inline Examples/VimeoUpload-iOS/VimeoUpload-iOS/Cells/VideoCell.swift]: Here too.

File: VimeoUpload.xcworkspace/contents.xcworkspacedata [hunk 5]
```diff
@@ -2,15 +2,15 @@
 <Workspace
```
Comment (supperspidey) [inline VimeoUpload.xcworkspace/contents.xcworkspacedata]: Sometimes I see `xcworkspacedata` and `xcsmblueprint` pop up. Do you know what those files are for?

File: VimeoUpload/UIKit/DescriptorKVObserver.swift [hunk 6]
```diff
@@ -86,37 +86,37 @@ public class DescriptorKVObserver: NSObject
     
     private func addObserversIfNecessary()
     {
-        if let descriptor = self.descriptor where self.observersAdded == false
+        if let descriptor = self.descriptor, self.observersAdded == false
         {
-            descriptor.progressDescriptor.addObserver(self, forKeyPath: self.dynamicType.StateKeyPath, options: .New, context: &self.stateKVOContext)
-            descriptor.progressDescriptor.addObserver(self, forKeyPath: self.dynamicType.ProgressKeyPath, options: .New, context: &self.progressKVOContext)
+            descriptor.progressDescriptor.addObserver(self, forKeyPath: type(of: self).StateKeyPath, options: .new, context: &self.stateKVOContext)
```
Comment (supperspidey) [inline VimeoUpload/UIKit/DescriptorKVObserver.swift]: Nice!

File: Examples/VimeoUpload-iOS-OldUpload/VimeoUpload-iOS-OldUpload/Cells/UploadCell.swift (hunk 1 above)
Comment (ghking) [inline Examples/VimeoUpload-iOS-OldUpload/VimeoUpload-iOS-OldUpload/Cells/UploadCell.swift]: lol who put these things in here

File: VimeoUpload.xcworkspace/contents.xcworkspacedata (hunk 5 above)
Comment (ghking) [inline VimeoUpload.xcworkspace/contents.xcworkspacedata]: Eh some Xcode metadata. I think they sometimes get changed when you modify the project structure.

**Comment by ghking** [comment]
@supperspidey Removed the `;`s, let me know if you're good to merge.

**Pull request merged**